### ISSUE TITLE
Improve AWS Codegen

### ISSUE THREAD

**Issue by bbernays** [cloudquery/cloudquery#2767]
Currently the codegen functionality only supports generating the resources (ie tables) it would help developers if codegen did the following tasks:
- Add function call to `plugins/source/aws/resources/plugin/tables.go`
- Add Client Definition to `plugins/source/aws/client/services.go`
- Add client to `Services` struct in `plugins/source/aws/client/client.go`
- Add client to `initServices` func in `plugins/source/aws/client/client.go`

This work would help reduce the complexity of adding a new resource by drastically reducing the number of steps. Reducing the number of steps will both save developer time, reduce the possibility of errors/panics as well as make the process of adding more resources more approachable to other developers

**Comment by bbernays** [comment]
@erezrokah - How much of the mock tests would you like to see generated? All of it? Some stubs? Some where in between?

**Comment by yevgenypats** [comment]
We need to be very careful with codegen doing too many things as it leads to complex developer experience and contribution which is then is backward of what we are trying to achieve.

The first thing I would start is tests which will catch  the issue such as missing function calls and so on because those will be needed in any case (if it is created via codegen or not).

Others improvements - it really depends on the value because if codegen works in all cases then great but if it become code that controls codegen then it is better to right the code without codegen. We saw this happening when we tried to generate aws `fetch` and `mock` functions. we shouldn't do that for sure as we tried that and it didn't go well (i.e amount of codegen code was bigger then the code itself but the complexity was much higher). Regarding the other suggestion things, really depends on each case. 

In any-case definitely tests and documentation first -> automation and features later.
